src/Symfony/Component/Validator/Constraints/Range.php

```diff
@@ -57,36 +60,62 @@ class Range extends Constraint
-            }
+        if (null === $this->min && null === $this->minPropertyPath && null === $this->max && null === $this->maxPropertyPath) {
+            throw new MissingOptionsException(sprintf('Either option "min", "minPropertyPath", "max" or "maxPropertyPath" must be given for constraint "%s".', __CLASS__), ['min', 'minPropertyPath', 'max', 'maxPropertyPath']);
+        }
 
-            if (isset($options['min']) && isset($options['max'])) {
-                $this->deprecatedMinMessageSet = isset($options['minMessage']);
-                $this->deprecatedMaxMessageSet = isset($options['maxMessage']);
+        if (null !== $this->min && null !== $this->minPropertyPath) {
+            throw new ConstraintDefinitionException(sprintf('The "%s" constraint requires only one of the "min" or "minPropertyPath" options to be set, not both.', static::class));
+        }
 
-                // BC layer, should throw a ConstraintDefinitionException in 6.0
-                if ($this->deprecatedMinMessageSet || $this->deprecatedMaxMessageSet) {
-                    trigger_deprecation('symfony/validator', '4.4', '"minMessage" and "maxMessage" are deprecated when the "min" and "max" options are both set. Use "notInRangeMessage" instead.');
-                }
-            }
+        if (null !== $this->max && null !== $this->maxPropertyPath) {
+            throw new ConstraintDefinitionException(sprintf('The "%s" constraint requires only one of the "max" or "maxPropertyPath" options to be set, not both.', static::class));
         }
 
-        parent::__construct($options);
+        if ((null !== $this->minPropertyPath || null !== $this->maxPropertyPath) && !class_exists(PropertyAccess::class)) {
+            throw new LogicException(sprintf('The "%s" constraint requires the Symfony PropertyAccess component to use the "minPropertyPath" or "maxPropertyPath" option.', static::class));
+        }
 
-        if (null === $this->min && null === $this->minPropertyPath && null === $this->max && null === $this->maxPropertyPath) {
-            throw new MissingOptionsException(sprintf('Either option "min", "minPropertyPath", "max" or "maxPropertyPath" must be given for constraint "%s".', __CLASS__), ['min', 'minPropertyPath', 'max', 'maxPropertyPath']);
+        if (null !== $this->min && null !== $this->max) {
+            $this->deprecatedMinMessageSet = isset($options['minMessage']) || null !== $minMessage;
+            $this->deprecatedMaxMessageSet = isset($options['maxMessage']) || null !== $maxMessage;
+
+            // BC layer, should throw a ConstraintDefinitionException in 6.0
+            if ($this->deprecatedMinMessageSet || $this->deprecatedMaxMessageSet) {
+                trigger_deprecation('symfony/validator', '4.4', '"minMessage" and "maxMessage" are deprecated when the "min" and "max" options are both set. Use "notInRangeMessage" instead.');
```

Comment:
I know that you didn't change that line, but 4.4 seems suspicious here. Should be 5.something I suppose

Reply:
4.4 is correct, see #37821 for context.